docs/src/main/asciidoc/mongodb-panache-guide.adoc

```diff
@@ -0,0 +1,492 @@
+* place your entity logic in static methods in your entities
+
+Then add the relevant configuration properties in `{config-file}`.
+
+[source,properties]
+--
+# configure the MongoDB client for a replica set of two nodes
+quarkus.mongodb.connection-string = mongodb://mongo1:27017,mongo2:27017
+# mandatory if you don't specify the name of the database using @MongoEntity
+quarkus.mongodb.database = person
+--
+
+The `quarkus.mongodb.database` property will be used by MongoDB with Panache to determine the name of the database where your entities will be persisted.
+
+For advanced configuration of the MongoDB client, you can follow the link:mongo-guide.html#configuring-the-mongodb-database[Configuring the MongoDB database].
+
+== Defining your entity
+
+To define a Panache entity, simply extend `PanacheMongoEntity`, annotate it with `@MongoEntity` and add your
+columns as public fields:
+
+[source,java]
+--
+@MongoEntity(collection="ThePerson")
+public class Person extends PanacheMongoEntity {
+    public String name;
+    public LocalDate birth;
+    public Status status;
+}
+--
+
+NOTE: annotating with `@MongoEntity` is optional, it allows to configure the name of the collection and the name of the database. Here the entity will be stored in the `ThePerson` collection instead of the default `Person` collection.
+
+MongoDB with Panache uses the link:https://mongodb.github.io/mongo-java-driver/3.10/bson/pojos/[PojoCodecProvider] to map your entities to MongoDB Document. 
+
+You will be allowed to use the following annotations to customize this mapping:
+
+- `@BsonId`: customize the id field, by default it's the field named `id` and of type `ObjectId`.
+- `@BsonProperty`: customize the serialized name of the field.
+- `@BsonIgnore`: ignore a field during the serialization.
+
+If you need to write accessors, you can:
+
+[source,java]
+--
+public class Person extends PanacheMongoEntity {
+    public String name;
+    public LocalDate birth;
+    public Status status;
+
+    // return name as uppercase in the model
+    public String getName(){
+      return name.toUppercase();
+    }
+
+    // store all names in lowercase in the DB
+    public String setName(String name){
+      this.name = name.toLowercase();
+    }
+}
+--
+
+And thanks to our field access rewrite, when your users read `person.name` they will actually call your `getName()` accessor, and similarly for field writes and the setter.
+This allows for proper encapsulation at runtime as all fields calls will be replaced by the corresponding getter/setter calls.
+
+== Most useful operations
+
+Once you have written your entity, here are the most common operations you will be able to do:
+
+[source,java]
+--
+// creating a person
+Person person = new Person();
+person.name = "Stef";
+person.birth = LocalDate.of(1910, Month.FEBRUARY, 1);
+person.status = Status.Alive;
+
+// persist it
+person.persist();
```

Comment:
We need to specify that entity needs to be explicitly updated yes. 
But per request of @emmanuelbernard I made the `persist()` method able to both insert and update (it's a replaceOne wiht upsert enabled). This is the last open point of my PR: whether or not I keep insert/update/persist or just persist ?

Reply:
OK, but that's orthogonal to making sure people are away that an "update" operation (whatever its name) is required because the entity is not "managed".

Reply:
I added the following note:
```
NOTE: The `persist()` method needs to be called each time you do a modification to your entity.
```

Reply:
I would put your note right below like this (if we go for a single `persist` method):

```java
person.status = Status.Dead;

// Your must call persist() again in order to send your entity modifications to MongoDB
person.persist();
```

This gives us a clearer workflow example.

Reply:
done